sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/sas/DataLakeServiceSasQueryParameters.java

```diff
@@ -0,0 +1,276 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.sas;
+
+import com.azure.storage.common.Utility;
+import com.azure.storage.common.implementation.Constants;
+import com.azure.storage.common.sas.BaseSasQueryParameters;
+import com.azure.storage.common.sas.SasIpRange;
+import com.azure.storage.common.sas.SasProtocol;
+import com.azure.storage.file.datalake.PathClientBuilder;
+import com.azure.storage.file.datalake.models.UserDelegationKey;
+
+import java.time.OffsetDateTime;
+import java.util.Map;
+
+/**
+ * Represents the components that make up an Azure Storage SAS' query parameters. This type is not constructed directly
+ * by the user; it is only generated by the {@link DataLakeServiceSasSignatureValues} type. Once generated, it can be
+ * set on a {@link PathClientBuilder} object to be constructed as part of a URL or it can be encoded into a
```

Comment:
It looks like the builder method actually accepts a string, not SasQueryParameters

Reply:
Fixed. We will need to take note to change this everywhere else as well